submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+    this.searchInput = document.querySelector('#nameInput');
+    this.startFilter();
+    this.sortByNameTypes = {
+      type1: 'nameAZ',
+      type2: 'nameZA',
+      type3: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      type1: 'age09',
+      type2: 'age90',
+      type3: 'ageAll',
+    };
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const activeRadioButtons = document.querySelectorAll(
+      'input[type=radio]:checked'
+    );
+    const checkedRadio = Array.from(activeRadioButtons).find(
+      node => node.name === nodeName
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSort(selected) {
+    const profileNames = document.querySelectorAll('.user-mail');
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let filter = target.value.toUpperCase();
+      profileNames.forEach(nameNode => {
+        if (nameNode.textContent.toLocaleUpperCase().startsWith(filter)) {
+          nameNode.closest('.user-card').hidden = false;
+        } else {
+          nameNode.closest('.user-card').hidden = true;
+        }
+      });
+    });
+
+    const sort = {
+      'sort-gender': () => this.sortByGender(selected.id),
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent('.first-name', selected.id, this.sortByNameTypes);
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent('.age', selected.id, this.sortByAgeTypes);
+      },
+      default: () => false,
+    };
+    return (sort[selected.name] || sort['default'])();
+  }
+
+  startFilter() {
+    this.filtersPanel.addEventListener('click', ({ target }) => {
+      if (target.type === 'radio' || target.type === 'text') {
+        loadMoreData.loadDataByScroll(true);
+        this.selectSort(target);
+      }
+    });
+  }
+
+  sortByGender(key) {
+    const cards = document.querySelectorAll('.user-card');
+    if (key === 'allGender') {
+      Array.from(cards).map(node => node.classList.remove('hidden'));
+    } else {
+      Array.from(cards).find(node => {
+        const currentGender = node.querySelector('.gender');
+        if (currentGender.textContent !== key) {
+          currentGender.closest('.user-card').classList.add('hidden');
+        } else {
+          currentGender.closest('.user-card').classList.remove('hidden');
+        }
+      });
+    }
+  }
+
+  insertSortedProfiles(sortedCards, param) {
+    sortedCards.forEach(card =>
+      this.profilesContainer.insertBefore(card, this.profilesContainer[param])
+    );
+  }
+
+  sortBySelector(selectorName) {
+    const cards = document.querySelectorAll('.user-card');
+    if (selectorName === 'all') {
+      return Array.from(cards).sort((a, b) => a.id.localeCompare(b.id));
+    }
+    return Array.from(cards).sort((a, b) =>
+      a
+        .querySelector(selectorName)
+        .textContent.localeCompare(b.querySelector(selectorName).textContent)
+    );
```

Comment:
Same, you changing DOM itself and you should change array of sources. And then render it again

Reply:
Got it! According to your recommendation I save data to array and then work with it, sorted and then render